content/articles/todolist-with-svelte-and-firebase/index.md

```diff
@@ -0,0 +1,1135 @@
+      border-radius: 50%;
+      padding: 1rem;
+      border: none;
+      background: #1890ff;
+      margin-right: 1rem;
+    }
+    .info {
+      p {
+        font-size: 1.4rem;
+        margin-bottom: 0.5rem;
+        &.cancel {
+          text-decoration: line-through;
+        }
+      }
+    }
+  }
+</style>
+```
+
+This component takes a `todo` as a prop and renders its name as well as stars to represent its priority. It also has a button that marks its `complete` or `incomplete` state.
+
+There's just one issue, Svelte cannot import `.svg` files by itself. We need to add packages to help us do that. Run
+
+```terminal
+yarn add -D rollup-plugin-svelte-svg
+```
+
+ and replace the `rollup.config.js` file with this:
+
+```js
+//CODE OMITTED FOR BREVITY
+import css from "rollup-plugin-css-only";
+
+//NEW LINE HERE
+import { svelteSVG } from "rollup-plugin-svelte-svg";
+
+
+//CODE OMITTED FOR BREVITY
+  plugins: [
+    //NEW LINE HERE
+    svelteSVG({
+      svgo: {},
+    }),
+    svelte({
+      preprocess: preprocess(),
+      compilerOptions: {
+        dev: !production,
+      },
+    }),
+//CODE OMITTED FOR BREVITY
+    
+```
+
+Running
+
+```terminal
+  yarn dev
+```
+
+would still throw an error as we currently have no assets to import.
+
+Create a folder named `assets` in the `src` directory and paste these three files:
+
+- cancel.svg
+  
+  ```svg
+  <?xml version="1.0" ?><svg viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg"><defs><style>.cls-1{fill:none;}</style></defs><title/><g data-name="Layer 2" id="Layer_2"><path d="M4,29a1,1,0,0,1-.71-.29,1,1,0,0,1,0-1.42l24-24a1,1,0,1,1,1.42,1.42l-24,24A1,1,0,0,1,4,29Z"/><path d="M28,29a1,1,0,0,1-.71-.29l-24-24A1,1,0,0,1,4.71,3.29l24,24a1,1,0,0,1,0,1.42A1,1,0,0,1,28,29Z"/></g><g id="frame"><rect class="cls-1" height="32" width="32"/></g></svg>
+  ```
+
+- star.svg
+  
+  ```svg
+  <?xml version="1.0" encoding="UTF-8"?><svg enable-background="new 0 0 47.94 47.94" version="1.1" viewBox="0 0 47.94 47.94" xml:space="preserve" xmlns="http://www.w3.org/2000/svg"><path d="m26.285 2.486 5.407 10.956c0.376 0.762 1.103 1.29 1.944 1.412l12.091 1.757c2.118 0.308 2.963 2.91 1.431 4.403l-8.749 8.528c-0.608 0.593-0.886 1.448-0.742 2.285l2.065 12.042c0.362 2.109-1.852 3.717-3.746 2.722l-10.814-5.685c-0.752-0.395-1.651-0.395-2.403 0l-10.814 5.685c-1.894 0.996-4.108-0.613-3.746-2.722l2.065-12.042c0.144-0.837-0.134-1.692-0.742-2.285l-8.749-8.528c-1.532-1.494-0.687-4.096 1.431-4.403l12.091-1.757c0.841-0.122 1.568-0.65 1.944-1.412l5.407-10.956c0.946-1.919 3.682-1.919 4.629 0z" fill="#ED8A19"/></svg>
+  ```
+
+- tick.svg
+  
+  ```svg
+  <?xml version="1.0" ?><svg id="Layer_1" style="enable-background:new 0 0 50 50;" version="1.1" viewBox="0 0 50 50" xml:space="preserve" fill='#fff' xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><g id="Layer_1_1_"><polygon points="47.293,6.94 14,40.232 2.707,28.94 1.293,30.353 14,43.06 48.707,8.353  "/></g></svg>
+  ```
```

Comment:
Let's import them from the GitHub repo. Need not specify them here. Add URLs instead